site/bulletin_board/routes/users.js

```
@@ -84,4 +84,19 @@ router.post('/', (req, res, next) => {
   });
 });
 
+/*--------changes password--------------*/
+router.post('/edit_password/', (req, res, next) => {
+  datasource.Editing_password(req.body, req.user, (result) => {
+    res.send(result);
+  });
+});
+router.get('/edit_password/', (req, res, next) => {
+  var profile = req.body;
+    res.render('user_edit_password');
+});
```

One suggestion for your consideration...

Right now, if the user is logged out and hits these endpoints (maybe their session expired?) I think you will see some server-side errors when `req.user` is unexpectedly undefined.

This is just another way of saying: "only signed-in users should be able to edit their profiles".

As-is, this is all safe from authorization issues because your first SQL statement is `SELECT * FROM Users WHERE id = ?` - which will probably fail when the user is undefined - but we could make this assumption explicit, automatically redirect to the login page, and return correct 500 errors if you were to add a router to handle `./password` which requires the user be signed-in.